File: frontend/src/types/project.ts
export interface Project {
  id: number;

  title: string;
  abstract: string;

  department: string;
  year: number;

  technologies: string;

  github_link?: string;

  status: "Pending" | "Approved" | "Rejected";

  guide_id: number;
  guide_name: string;

  student_name: string;
}

File: backend/app/schemas/project.py
from pydantic import BaseModel
from typing import Literal

class ProjectCreate(BaseModel):
    title: str
    abstract: str
    department: str
    year: int
    guide_id : int
    technologies: str
    github_link: str | None = None


class ProjectResponse(BaseModel):
    id: int

    title: str
    abstract: str

    department: str
    year: int

    technologies: str

    github_link: str | None

    status: str

    guide_id: int

    guide_name: str

    student_name: str

    class Config:
        from_attributes = True

class ProjectStatusUpdate(BaseModel):
    status: Literal["Pending", "Approved", "Rejected"]

class ProjectEdit(BaseModel):
    title: str
    abstract: str
    department: str
    year: int
    guide_id : int
    technologies: str
    github_link: str | None = None

class FilterOptionsResponse(BaseModel):
    departments: list[str]
    years: list[int]
    guides: list[str]
    technologies: list[str]